src/test/java/teammates/test/pageobjects/GoogleLoginPage.java

```diff
@@ -84,24 +88,21 @@ private void handleApprovalPageIfAny() {
         waitForRedirectIfAny();
         boolean isPageRequestingAccessApproval = getPageSource().contains(EXPECTED_SNIPPET_APPROVAL);
         if (isPageRequestingAccessApproval) {
-            click(By.id("persist_checkbox"));
+            markCheckBoxAsChecked(browser.driver.findElement(By.id("persist_checkbox")));
             click(By.id("approve_button"));
             waitForPageToLoad();
         }
     }
 
     private void waitForRedirectIfAny() {
-        By metaRefreshBy = By.cssSelector("meta[http-equiv='refresh']");
-        if (isElementPresent(metaRefreshBy)) {
-            waitForElementToDisappear(metaRefreshBy);
-            waitForPageToLoad();
-        }
-
-        By noScriptBy = By.tagName("noscript");
-        if (isElementPresent(noScriptBy)) {
-            waitForTextContainedInElementAbsence(noScriptBy, "&lt;meta http-equiv=\"refresh\"");
-            waitForPageToLoad();
-        }
+        String loginRedirectUrl = TestProperties.TEAMMATES_URL + "/_";
```

Comment:
In that case, can make the URL pattern stricter? e.g. `/_ah/conflogin`. The current one should work and unlikely to cause false positives, but if possible to make it stricter, it will be clearer as well.